### PR TITLE
[trivial] Fix typos in comments

### PR DESCRIPTION
**Pull request by practicalswift**

### CODE REVIEW

File: src/secp256k1/src/java/org/bitcoin/NativeSecp256k1.java [hunk 1]
```diff
@@ -359,8 +359,8 @@ public static long cloneContext() {
     /**
      * libsecp256k1 create ECDH secret - constant time ECDH calculation
      *
-     * @param seckey byte array of secret key used in exponentiaion
-     * @param pubkey byte array of public key used in exponentiaion
+     * @param seckey byte array of secret key used in exponentiation
```
Comment (paveljanik) [inline src/secp256k1/src/java/org/bitcoin/NativeSecp256k1.java]: Please report this upstream and remove it from here.

File: src/leveldb/util/env_win.cc [hunk 2]
```diff
@@ -1,7 +1,7 @@
 // This file contains source that originates from:
 // http://code.google.com/p/leveldbwin/source/browse/trunk/win32_impl_src/env_win32.h
 // http://code.google.com/p/leveldbwin/source/browse/trunk/win32_impl_src/port_win32.cc
-// Those files dont' have any explict license headers but the 
+// Those files don't have any explicit license headers but the
```
Comment (paveljanik) [inline src/leveldb/util/env_win.cc]: Please report this upstream and remove it from here.

File: src/leveldb/port/port_posix.h [hunk 3]
```diff
@@ -71,7 +71,7 @@
 
 #if defined(OS_ANDROID) && __ANDROID_API__ < 9
 // fdatasync() was only introduced in API level 9 on Android. Use fsync()
-// when targetting older platforms.
+// when targeting older platforms.
```
Comment (paveljanik) [inline src/leveldb/port/port_posix.h]: Please report this upstream and remove it from here.

File: build-aux/m4/ax_boost_base.m4 [hunk 4]
```diff
@@ -10,7 +10,7 @@
 #
 #   Test for the Boost C++ libraries of a particular version (or newer)
 #
-#   If no path to the installed boost library is given the macro searchs
+#   If no path to the installed boost library is given the macro searches
```
Comment (paveljanik) [inline build-aux/m4/ax_boost_base.m4]: Please report this upstream and remove it from here.

**Comment by paveljanik** [comment]
Good catches! Why do you have [gardening] in the commit message?

File: doc/Doxyfile [hunk 5]
```diff
@@ -1486,7 +1486,7 @@ EXT_LINKS_IN_WINDOW    = NO
 
 FORMULA_FONTSIZE       = 10
 
-# Use the FORMULA_TRANPARENT tag to determine whether or not the images
+# Use the FORMULA_TRANSPARENT tag to determine whether or not the images
```
Comment (paveljanik) [inline doc/Doxyfile]: Can you please also report this typo also to doxygen (src/config.xml)?

**Comment by practicalswift** [comment]
@paveljanik Fixed and pushed!

The `[gardening]` part is a long standing tradition from [my contributions to the `apple/swift` repo](https://github.com/apple/swift/pulls?utf8=%E2%9C%93&q=is%3Apr%20author%3Apracticalswift%20is%3Amerged) :-)   It simply means that the commit contains only hopefully non-controversial non-functional changes such as typo fixes, general cleanups, etc.

File: doc/Doxyfile (hunk 5 above)
Comment (practicalswift) [inline doc/Doxyfile]: Sure! Will do!

**Comment by paveljanik** [comment]
Concept ACK (will be read-ACK when you change/remove the [gardening]. We use 'trivial' here.

I like gardening BTW - I prefer/collect magnolias ;-)

**Comment by practicalswift** [comment]
@paveljanik Fixed and pushed! :-)

Thanks for the quick review! Two magnolia leafs for you: 🍃

File: doc/Doxyfile (hunk 5 above)
Comment (fanquake) [inline doc/Doxyfile]: I think this can just be sent upstream as well. It'll be pulled in if/when we regenerate the Doxyfile next.

**Comment by fanquake** [comment]
utACK post nits being addressed.

File: src/tinyformat.h [hunk 6]
```diff
@@ -483,7 +483,7 @@ namespace detail {
 
 // Type-opaque holder for an argument to format(), with associated actions on
 // the type held as explicit function pointers.  This allows FormatArg's for
-// each argument to be allocated as a homogenous array inside FormatList
+// each argument to be allocated as a homogeneous array inside FormatList
```
Comment (MarcoFalke) [inline src/tinyformat.h]: tinyformat is from upstream as well

**Comment by practicalswift** [comment]
The changes to `doc/Doxyfile` and `src/tinyformat.h` are now excluded as per the suggestions from @fanquake and @MarcoFalke :-)

Please let me know if any further

**Comment by fanquake** [comment]
ACK f2a4484

File: src/base58.h [hunk 7]
```diff
@@ -147,7 +147,7 @@ template<typename K, int Size, CChainParams::Base58Type Type> class CBitcoinExtK
     K GetKey() {
         K ret;
         if (vchData.size() == Size) {
-            //if base58 encouded data not holds a ext key, return a !IsValid() key
+            //if base58 encoded data not holds a ext key, return a !IsValid() key
```
Comment (sipa) [inline src/base58.h]: This still doesn't sound like english.

Comment (fanquake) [inline src/base58.h]: right. *data does not s/holds/hold

File: src/txmempool.h [hunk 8]
```diff
@@ -340,7 +340,7 @@ struct TxMempoolInfo
  * Transactions are added when they are seen on the network (or created by the
  * local node), but not all transactions seen are added to the pool. For
  * example, the following new transactions will not be added to the mempool:
- * - a transaction which doesn't make the mimimum fee requirements.
+ * - a transaction which doesn't make the minimum fee requirements.
```
Comment (sipa) [inline src/txmempool.h]: doesn't meet?

File: src/base58.h (hunk 7 above)
Comment (practicalswift) [inline src/base58.h]: Good point!

I'm not familiar with this piece of the code - any suggestions on what the intention was here?

> If the base58 encoded data does not hold an ext key, return an !IsValid() key

Could that be the correct guess? :-)

File: src/txmempool.h (hunk 8 above)
Comment (practicalswift) [inline src/txmempool.h]: Fixed and pushed! :-)

**Comment by practicalswift** [comment]
@sipa & @fanquake Do you have input on the outstanding question on how to interpret the line `if base58 encoded data`?

Let me know if that change needs more time for review - if so I can simply open a separate PR for that one to allow this one to be merged with the approved changes only :-)

**Comment by fanquake** [comment]
@practicalswift See my comment inline: https://github.com/bitcoin/bitcoin/pull/9510#discussion_r95788193, your interpretation looks correct. A second PR is definitely not needed here.

**Comment by practicalswift** [comment]
@fanquake Fixed! Everything OK now? :-)

**Comment by practicalswift** [comment]
Merge conflict resolved. This one should be ready for merge AFAIK :-)

**Comment by paveljanik** [comment]
ACK https://github.com/bitcoin/bitcoin/pull/9510/commits/cc16d99f1dc8305b1b255f1cc0f2b1516aa77ed0

**Comment by MarcoFalke** [comment]
utACK cc16d99f1dc8305b1b255f1cc0f2b1516aa77

**Comment by practicalswift** [comment]
@MarcoFalke Thanks for merging! Another round of typo fixes in PR #9651 :-)

File: src/init.cpp [hunk 9]
```diff
@@ -1553,7 +1553,7 @@ bool AppInitMain(boost::thread_group& threadGroup, CScheduler& scheduler)
     }
 
     if (chainparams.GetConsensus().vDeployments[Consensus::DEPLOYMENT_SEGWIT].nTimeout != 0) {
-        // Only advertize witness capabilities if they have a reasonable start time.
```
Comment (rebroad) [inline src/init.cpp]: this probably was not a typo. it's how it's spelled in some countries.

Comment (practicalswift) [inline src/init.cpp]: @rebroad Do you know in what countries? As I've understood it "advertise" is the form used in both US and UK, but there are more English speaking countries than that obviously :-)

Comment (rebroad) [inline src/init.cpp]: @practicalswift I apologise, you are correct!

**Pull request merged**